src/components/input/_input.scss

```diff
@@ -23,18 +31,18 @@
     appearance: none;
   }
 
-  .govuk-c-input::-webkit-outer-spin-button,
-  .govuk-c-input::-webkit-inner-spin-button {
+  .govuk-c-input__input::-webkit-outer-spin-button,
+  .govuk-c-input__input::-webkit-inner-spin-button {
     margin: 0;
     -webkit-appearance: none;
   }
 
-  .govuk-c-input[type="number"] {
+  .govuk-c-input__input[type="number"] {
     -moz-appearance: textfield;
   }
 
-  .govuk-c-input--error {
-    border: $govuk-border-width-error solid $govuk-error-colour;
+  .govuk-c-input__input--error {
```

Comment:
Do we have other places where we're applying a modifier to an element like this? This is probably something we need to try and be consistent on, as another approach would be to do `.govuk-c-input--error .govuk-c-input__input` – i.e. add the modifier to the block, and style elements as children.

Reply:
Thanks @36degrees. This was trying to be consistent with `govuk-c-date-input__input` and `govuk-c-date-input__item--year` but it's no longer a thing as everything uses `form-o-group`.

We should possibly think about this so that we can provide guidance to contributors who want to create new components.